adapters/v1/grype.go

```diff
@@ -0,0 +1,147 @@
+	dbConfig := db.Config{
+		DBRootDir:  "grypedb",
+		ListingURL: "https://toolbox-data.anchore.io/grype/databases/listing.json",
+	}
+	g := &GrypeAdapter{
+		dbConfig: dbConfig,
+	}
+	err := g.UpdateDB(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return g, nil
+}
+
+func (g *GrypeAdapter) CreateRelevantCVE(ctx context.Context, cve, cvep domain.CVE) (domain.CVE, error) {
+	// TODO implement me
+	panic("implement me")
+}
+
+func (g *GrypeAdapter) DBVersion() string {
+	return g.dbStatus.Checksum
+}
+
+func (g *GrypeAdapter) Ready() bool {
+	return g.dbStatus.Err != nil
+}
+
+func getMatchers() []matcher.Matcher {
+	return matcher.NewDefaultMatchers(
+		matcher.Config{
+			Java: java.MatcherConfig{
+				ExternalSearchConfig: java.ExternalSearchConfig{MavenBaseURL: "https://search.maven.org/solrsearch/select"},
+				UseCPEs:              true,
+			},
+			Ruby:       ruby.MatcherConfig{UseCPEs: true},
+			Python:     python.MatcherConfig{UseCPEs: true},
+			Dotnet:     dotnet.MatcherConfig{UseCPEs: true},
+			Javascript: javascript.MatcherConfig{UseCPEs: true},
+			Golang:     golang.MatcherConfig{UseCPEs: true},
+			Stock:      stock.MatcherConfig{UseCPEs: true},
+		},
+	)
+}
+
+func (g *GrypeAdapter) ScanSBOM(ctx context.Context, sbom domain.SBOM) (domain.CVE, error) {
+	g.mu.RLock()
+	defer g.mu.RUnlock()
+	ctx, span := otel.Tracer("").Start(ctx, "ScanSBOM")
+	defer span.End()
+	s, _, err := syft.Decode(bytes.NewReader(sbom.Content))
+	if err != nil {
+		return domain.CVE{}, err
+	}
+	packages := pkg.FromCatalog(s.Artifacts.PackageCatalog, pkg.SynthesisConfig{})
+	if err != nil {
+		return domain.CVE{}, err
+	}
+	pkgContext := pkg.Context{
+		Source: &s.Source,
+		Distro: s.Artifacts.LinuxDistribution,
+	}
+	vulnMatcher := grype.VulnerabilityMatcher{
+		Store:    *g.store,
+		Matchers: getMatchers(),
+	}
+	remainingMatches, ignoredMatches, err := vulnMatcher.FindMatches(packages, pkgContext)
+	if err != nil {
+		return domain.CVE{}, err
+	}
+	// generate JSON
+	presenterConfig := models.PresenterConfig{
+		Matches:          *remainingMatches,
+		IgnoredMatches:   ignoredMatches,
+		Packages:         packages,
+		Context:          pkgContext,
+		MetadataProvider: g.store,
+		SBOM:             s,
+		AppConfig:        nil,
+		DBStatus:         g.dbStatus,
+	}
+	presenter := json.NewPresenter(presenterConfig)
+	var buf bytes.Buffer
+	err = presenter.Present(&buf)
+	if err != nil {
+		return domain.CVE{}, err
+	}
+	return domain.CVE{
+		ImageID:            sbom.ImageID,
+		SBOMCreatorVersion: sbom.SBOMCreatorVersion,
+		CVEScannerVersion:  g.Version(),
+		CVEDBVersion:       g.DBVersion(),
+		Content:            buf.Bytes(),
+	}, nil
+}
+
+func (g *GrypeAdapter) UpdateDB(ctx context.Context) error {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+	ctx, span := otel.Tracer("").Start(ctx, "UpdateDB")
+	defer span.End()
+	var err error
+	g.store, g.dbStatus, g.dbCloser, err = grype.LoadVulnerabilityDB(g.dbConfig, true)
```

Comment:
issue: `LoadVulnerabilityDB` visits the `dbConfig.ListingURL`

This makes a live request, and if we exceed our quota or something like that, our tests will fail or will be flaky otherwise. Please see the comment regarding `dbConfig`.